R-package/R/lightgbm.R

```diff
@@ -92,6 +92,18 @@ NULL
 #' @inheritParams lgb_shared_params
 #' @param label Vector of labels, used if \code{data} is not an \code{\link{lgb.Dataset}}
 #' @param weight vector of response values. If not NULL, will set to dataset
+#' @param nthreads Number of parallel threads to use. For best speed, this should be set to the number of
+#'                 physical cores in the CPU - in a typical x86-64 machine, this corresponds to half the
+#'                 number of maximum threads (e.g. `nthreads = max(parallel::detectCores() / 2L, 1L)` as
+#'                 a shorthand for the optimal value).
```

Comment:
```suggestion
#' @param nthreads Number of parallel threads to use. For best speed, this should be set to the number of
#'                 physical cores in the CPU - (e.g. \code{parallel::detectCores(logical = FALSE)})
```

As mentioned in https://github.com/microsoft/LightGBM/issues/4192#issuecomment-822023560, it's possible to get this information directly from `parallel::detectCores()` without using any heuristics. Please refer to that in the documentation instead.